test/id.test.js

```diff
@@ -247,164 +153,95 @@ describe('mongodb default id name', function() {
-        name: String,
-      },
-    );
-
-    beforeEach(function(done) {
-      User.deleteAll(done);
-    });
-
-    it('should coerce to ObjectID', async function() {
-      const user = await User.create({id: objectIdLikeString, name: 'John'});
-      user.id.should.be.an.instanceOf(ds.ObjectID);
-    });
-
-    it('should find model with ObjectID id', async function() {
-      const user = await User.create({name: 'John'});
-      user.id.should.be.an.instanceOf(ds.ObjectID);
-      const found = await User.findById(user.id);
-      found.toObject().name.should.equal('John');
-    });
-
-    it('should find model with ObjectID-like id', async function() {
-      const user = await User.create({id: objectIdLikeString, name: 'John'});
-      user.id.should.be.an.instanceOf(ds.ObjectID);
-      user.id.should.eql(ObjectID(objectIdLikeString));
-      const found = await User.findById(objectIdLikeString);
-      found.toObject().name.should.equal('John');
-    });
+describe('autogenerated ids', function() {
+  const Customer = ds.createModel(
+    'Customer2',
+    {name: String, emails: [String], age: Number},
+    {forceId: false},
+  );
 
-    it('should find model with string id', async function() {
-      const user = await User.create({id: 'a', name: 'John'});
-      user.id.should.be.an.instanceOf(String);
-      user.id.should.equal('a');
-      const found = await User.findById('a');
-      found.toObject().name.should.equal('John');
-    });
+  before(function(done) {
+    Customer.deleteAll(done);
   });
 
-  context('set to true', function() {
-    const User = ds.createModel(
-      'user2',
-      {
-        id: {type: String, id: true},
-        name: String,
-      },
-      {strictObjectIDCoercion: true},
-    );
-
-    beforeEach(function(done) {
-      User.deleteAll(done);
-    });
-
-    it('should not coerce to ObjectID', async function() {
-      const user = await User.create({id: objectIdLikeString, name: 'John'});
-      user.id.should.equal(objectIdLikeString);
-    });
-
-    it('should not find model with ObjectID id', async function() {
-      const user = await User.create({name: 'John'});
-      const found = await User.findById(user.id);
-      (found === null).should.be.true();
-    });
+  it('should not allow value for autogenerated id', async function() {
+    await Customer.create({
+      id: 1,
+      name: 'John1',
+      emails: ['john@x.com', 'john@y.com'],
+      age: 30,
+    }).should.be.rejectedWith(/Cannot specify value for autogenerated id/);
+  });
+});
 
-    it('should find model with ObjectID-like string id', async function() {
-      const user = await User.create({id: objectIdLikeString, name: 'John'});
-      user.id.should.not.be.an.instanceOf(ds.ObjectID);
-      user.id.should.eql(objectIdLikeString);
-      const found = await User.findById(objectIdLikeString);
-      found.toObject().name.should.equal('John');
-    });
+describe('non-generated ObjectId id', function() {
```

Comment:
Can we also have tests for non-autogenerated non-object-like string/number type id?

Reply:
There are covered in other tests like https://github.com/strongloop/loopback-connector-mongodb/blob/64351c43e36c62699b3ecc79fe4844af4e66b1d7/test/id.test.js#L17 and https://github.com/strongloop/loopback-connector-mongodb/blob/major/string-objectid/test/mongodb.test.js#L255.